airflow/configuration.py

```diff
@@ -398,8 +400,11 @@ def as_dict(
                     opt = opt.replace('%', '%%')
                 if display_source:
                     opt = (opt, 'env var')
-                cfg.setdefault(section.lower(), OrderedDict()).update(
-                    {key.lower(): opt})
+
+                section = section.lower()
+                if section != 'kubernetes_environment_variables':
```

Comment:
Why are we treating this one section differently? At the _very_ least we need a code comment here saying why, but ideally I'd like to not have any special case here if we can avoid it.

Reply:
This section needs to be case sensitive because airflow only parse environment variables starts with `AIRFLOW__` not `airflow__`. I also think this is a bit hacky.

We can also change it to not call `.lower` on all keys so we don't have to make `kubernetes_environment_variables` a special section. The catch is it will break existing users who depend on environment variables being case insensitive. i.e. you will have to inject environment variables like `AIRFLOW__FOO_bar=abc`.

Let me know what's your thought on this so I can go ahead and update this part of the code :)